unityfiles/Assets/Scripts/Game/Menu/PauseMenu.cs

```diff
@@ -0,0 +1,122 @@
+﻿using System;
+using UnityEngine;
+using UnityEngine.UI;
+
+public class PauseMenu : MonoBehaviour {
+    [SerializeField]
+    private GameObject pauseMenuCanvas;
+
+    [SerializeField]
+    private GameObject confirmDialog;
+
+    private Boolean isPaused = false;
+
+    private Button continueButton;
+    private static readonly int PAUSE = 0;
+    private static readonly float PLAY = 1;
+
+    private void Awake() {
+        CheckForMissingComponents();
+        
+        DeactivateMenu(); 
+    }
+
+    private void Start() {
+        SubscribeToEvents();
+    }
+
+    // Update is called once per frame
+    void Update() {
+        if (Input.GetButtonDown("Cancel")) {
+            if (confirmDialog.activeSelf) {
+                DeactivateConfirmDialog();
+            } else {
+                TogglePause();
+            }
+        }
+    }
+
+    /// <summary>
+    /// Toggles the pause state.
+    /// Shows pause menu if paused
+    /// </summary>
+    private void TogglePause() {
+        isPaused = !isPaused;
+
+        if (isPaused) {
+            PauseGame();
+        } else {
+            ResumeGame();
+        }
+    }
+
+    private void OnDestroy() {
+        UnsubscribeFromEvents();
+    }
+
+    private void UnsubscribeFromEvents() {
+        LevelManager.LevelStateChangeEvent -= ChangeGameState();
+    }
```

Comment:
MÅ si at jeg er sterkt uenig i at du bruker Level manager som en pass trogh for og få eventen din hit. hele meningen med et lukket system er at alle eventer hånteres I plassen eventhandleren er. hvis du trenger og håntere en event her jør det orgentlig og lag en egen handler her